### PR TITLE
chore: integrate Uffizzi PR Environments

### PR DESCRIPTION
**Pull request by waveywaves**
## Summary

Add support for creating pull request environment on every PR using Uffizzi.
- Build an image for every PR build in github actions
- Uffizzi deploy github action which uses the built image to deploy the PR Environment
- Docker compose template with uses postgres instance for every PR environment

## Testing

Tested this PR locally and on my fork of the repo https://github.com/waveywaves/logto/pull/2

## Additional information for maintainers
I want to invite the maintainers of Logto to the Uffizzi project created for them. The PR setups up PR Environments for Logto to ease development and reviews. Reviewers should be able to see a preview of the PR as a hosted service. The URL of the service will be provided in a comment on the PR itself. I want to invite the maintainers of Logto to the uffizzi project to get an idea of what the platform looks like. Uffizzi is free for Logto.

### CODE REVIEW

**Comment by waveywaves** [comment]
For this initially I would require the mail address of the developers of Logto to invite to the project.
cc @gao-sun @simeng-li @charIeszhao

File: uffizzi/docker-compose.uffizzi.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This compose file is for demonstration only, do not use in prod.
```
Comment (charIeszhao) [inline uffizzi/docker-compose.uffizzi.yml:1]: Why do we have both the dockerfile and docker-compose yml?

File: uffizzi/docker-compose.uffizzi.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    build:
+      context: ../.
+      dockerfile: ./uffizzi/Dockerfile
+    ports:
+      - 3001:3001
+    environment:
+      ALL_YES: 1
+      NO_INQUIRY: 0
+      TRUST_PROXY_HEADER: 1
+      DB_URL_DEFAULT: postgres://postgres:p0stgr3s@localhost:5432
```
Comment (charIeszhao) [inline uffizzi/docker-compose.uffizzi.yml:22]: Is it even possible to reuse our dockerfile in project root, and set the endpoint env here? Like:

```
build:
  dockerfile: ../Dockerfile
...
environment:
  ...
  ENDPOINT: $UFFIZZI_URL
```

Comment (charIeszhao) [reply uffizzi/docker-compose.uffizzi.yml:22]: You can refer to `.gitpod.yml` to see how it works for GitPod.

File: uffizzi/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+FROM node:16-alpine as app
+WORKDIR /etc/logto
+COPY --from=builder /etc/logto .
+
+COPY ./uffizzi/docker-entrypoint.sh .
+RUN chmod +x docker-entrypoint.sh
+
+EXPOSE 3001
+ENV NO_INQUIRY true
+ENTRYPOINT ["./docker-entrypoint.sh"]
```
[the remark this answers is not in the document]
Comment (waveywaves) [reply uffizzi/Dockerfile:33]: Not currently, we will review this after UFFIZZI_URL ingress maps are supported https://github.com/UffizziCloud/uffizzi_app/issues/251

File: uffizzi/docker-compose.uffizzi.yml (hunk 2 above)
Comment (waveywaves) [reply uffizzi/docker-compose.uffizzi.yml:22]: I will give it a go, this is something I can help with for sure

**Comment by gao-sun** [comment]
@charIeszhao

**Review by charIeszhao — APPROVED** [review]

**Pull request merged**